### PR TITLE
Feature/7 create user

### PR DESCRIPTION
**Pull request by alvgonfri**
Clinic admins can create users now.

### CODE REVIEW

File: src/controllers/userController.js [hunk 1]
```diff
@@ -4,7 +4,106 @@ import { redisClient } from '../config/redis.js';
...
+        method: req.method,
+        url: req.originalUrl,
+        userId: decoded.userId,
+        ip: req.ip,
+      });
+      return res.status(403).json({
+        message:
+          'You do not have permission to create a new user. Only clinic administrators can create new users.',
+      });
+    }
```
[the remark this answers is not in the document]
Comment (alvgonfri) [reply src/controllers/userController.js:35]: That's a good idea in order to reuse code.

File: src/controllers/userController.js [hunk 2]
```diff
@@ -4,7 +4,106 @@ import { redisClient } from '../config/redis.js';
...
+
+    if (!username || !password) {
+      logger.warn('Missing required fields: username or password', {
+        method: req.method,
+        url: req.originalUrl,
+        ip: req.ip,
+      });
+      return res.status(400).json({
+        message: 'Username and password are required fields.',
+      });
```
[the remark this answers is not in the document]
Comment (alvgonfri) [reply src/controllers/userController.js:47]: I will implement this improvement, too.

**Comment by AntonioRodriguezRuiz** [comment]
image
Not covered lines in userController relate to server errors

**Review by AntonioRodriguezRuiz — APPROVED** [review]
lgtm

**Pull request merged**